lib/ansible/modules/windows/win_domain_computer.ps1

```diff
@@ -0,0 +1,183 @@
+#!powershell
+#
+# Copyright: (c) 2017, AMTEGA - Xunta de Galicia
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+
+# ------------------------------------------------------------------------------
+$ErrorActionPreference = "Stop"
+
+# Preparing result
+$result = @{}
+Set-Attr $result "changed" $false
+
+# Parameter ingestion
+$params = Parse-Args $args -supports_check_mode $true
+
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -default $false | ConvertTo-Bool
+$diff_support = Get-AnsibleParam -obj $params -name "_ansible_diff" -default $false | ConvertTo-Bool
+
+# ------------------------------------------------------------------------------
+Function Get-DesiredState($params) {
+  $name = Get-AnsibleParam -obj $params -name "name" -failifempty $true -resultobj $result
+  $sam_account_name = Get-AnsibleParam -obj $params -name "sam_account_name" -default "$name$"
+  If (-not $sam_account_name.EndsWith("$")) {
+    Fail-Json $result "sam_account_name must end in $"
+  }
+  $enabled = Get-AnsibleParam -obj $params -name "enabled" -default $true | ConvertTo-Bool
+  $description = Get-AnsibleParam -obj $params -name "description" -default "" -resultobj $result
+  $state =(Get-AnsibleParam -obj $params -name "state" -ValidateSet "Present","Absent" -default "Present").ToLower()
+  If ($state -eq "present") {
+    $dns_hostname = Get-AnsibleParam -obj $params -name "dns_hostname" -failifempty $true -resultobj $result
+    $ou = Get-AnsibleParam -obj $params -name "ou" -failifempty $true -resultobj $result
+    $distinguished_name = "CN=$name,$ou"
+
+    $desired_state = @{
+      name = $name
+      sam_account_name = $sam_account_name
+      dns_hostname = $dns_hostname
+      ou = $ou
+      distinguished_name = $distinguished_name
+      description = $description
+      enabled = $enabled
+      state = $state
+    }
+  } Else {
+    $desired_state = @{
+      name = $name
+      state = $state
+    }
+  }
+
+  Return $desired_state
+}
+# ------------------------------------------------------------------------------
+Function Get-InitialState($desired_state) {
+  # Test computer exists
+  $computer = Try { Get-ADComputer `
```

Comment:
You are probably better off doing `Get-ADComputer .... -ErrorAction SilentlyContinue` instead of try/catching it. It's easier to ready that way in my mind but I'll leave it up to you how you want to proceed.